file/src/main/scala/akka/stream/alpakka/file/scaladsl/LogRotatorSink.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.file.scaladsl
+
+import java.nio.file.{Path, StandardOpenOption}
+
+import akka.Done
+import akka.stream._
+import akka.stream.scaladsl.{FileIO, Source}
+import akka.stream.stage._
+import akka.util.ByteString
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{Future, Promise}
+import scala.util.{Failure, Success}
+
+object LogRotatorSink {
+  def apply(functionGeneratorFunction: () => ByteString => Option[Path],
+            fileOpenOptions: Set[StandardOpenOption] = Set(StandardOpenOption.APPEND, StandardOpenOption.CREATE))
+    : LogRotatorSink =
```

Comment:
return type can be more secretive, `Sink[ByteString, Future[Done]]`, that is good as it does not lock down future changes, there is anyways no specific public API on `LogRotatorSink`.